### PR TITLE
Adds support of user layouts to benchmark_app

### PR DESCRIPTION
**Pull request by nkogteva**
There are list of issues connected to handling of inputs layout (30571, 43975) in benchmark_app.

Benchmark_app relies on layouts reported by IE. Since there are still open questions about layouts in IE, this MR introduce new parameter in benchmark_app where user can specify layouts for model inputs. This parameter are handled inside tool and doesn't affect IE.

Base on this layout batch size can be determined correctly (`N` dimension) and whether image is required on input or not (for 'NHWC' cases).

Usage sample:
```
-layout="[NHWC]"
```
or
```
-layout="input0[NC],input1[CN]"
```

### CODE REVIEW

File: inference-engine/samples/benchmark_app/benchmark_app.hpp [hunk 1]
```diff
@@ -215,6 +221,7 @@ static void showUsage() {
     std::cout << "    -t                        " << execution_time_message << std::endl;
     std::cout << "    -progress                 " << progress_message << std::endl;
     std::cout << "    -shape                    " << shape_message << std::endl;
+    std::cout << "    -layout                   " << layout_message << std::endl;
```
Comment (ilya-lavrenov) [inline inference-engine/samples/benchmark_app/benchmark_app.hpp:224]: can we have request for output layout? `compile_tool` has:
```
    -ip                          <value>     Optional. Specifies precision for all input layers of the network.
    -op                          <value>     Optional. Specifies precision for all output layers of the network.
    -iop                        "<value>"    Optional. Specifies precision for input and output layers by name.
                                             Example: -iop "input:FP16, output:FP16".
                                             Notice that quotes are required.
                                             Overwrites precision from ip and op options for specified layers.
    -il                          <value>     Optional. Specifies layout for all input layers of the network.
    -ol                          <value>     Optional. Specifies layout for all input layers of the network.
    -iol                        "<value>"    Optional. Specifies layout for input and output layers by name.
                                             Example: -iol "input:NCHW, output:NHWC".
                                             Notice that quotes are required.
                                             Overwrites layout from il and ol options for specified layers.
```
May be we can have the same notation which is more future proff?

Comment (nkogteva) [reply inference-engine/samples/benchmark_app/benchmark_app.hpp:224]: @ilya-lavrenov Format of new parameter is aligned with existed `shape` parameter. For example:
```
-shape "image_info[2,3],image_tensor[2,3,200,200]" -layout "image_info[NC],image_tensor[NCHW]"
```

I think it is better to be aligned inside one tool.

File: inference-engine/samples/benchmark_app/utils.hpp [hunk 2]
```diff
@@ -8,15 +8,119 @@
...
+        bool isImage() const;
+        bool isImageInfo() const;
+        size_t getDimentionByLayout(char character) const;
+        size_t width() const;
+        size_t height() const;
+        size_t channels() const;
+        size_t batch() const;
+        size_t depth() const;
+    };
+    using InputsInfo = std::map<std::string, InputInfo>;
```
Comment (ilya-lavrenov) [inline inference-engine/samples/benchmark_app/utils.hpp:25]: looks strange that **IE sample** use new structure different from IE ones. As far as I can see, you need only new methods, can such methods be outside of the class and work on IE structures?

Comment (nkogteva) [reply inference-engine/samples/benchmark_app/utils.hpp:25]: @ilya-lavrenov Having own duplicate structure is main goal of this MR.  By default, we still working with IE information, but in some cases IE gives wrong data and in this case we ask user set appropriate parameters and we need to keep this provided data somehow. Nothing should be changed in IE when user set input layouts except interpretation of them from application side.

**Review by akuporos — APPROVED** [review]
changes in python benchmark app LGTM except code style issues

**Review by ilya-lavrenov — APPROVED** [review]

File: tools/benchmark/utils/utils.py [hunk 3]
```diff
@@ -227,33 +226,105 @@ def get_command_line_arguments(argv):
...
+
+    @property
+    def batch(self):
+        return self.getDimentionByLayout("N")
+
+    @property
+    def depth(self):
+        return self.getDimentionByLayout("D")
+
+def getInputsInfo(shape_string, layout_string, batch_size, input_info):
```
[the remark this answers is not in the document]
Comment (nkogteva) [reply tools/benchmark/utils/utils.py:289]: @akuporos All functions in benchmark_app utils are CamelCase. The same style I see in common part of samples. Let's resolve style questions separately.

**Review by gladilov-gleb — APPROVED** [review]

**Pull request merged**